src/librustc_mir/borrow_check/nll/region_infer/error_reporting/mod.rs

```diff
@@ -120,19 +122,50 @@ impl<'tcx> RegionInferenceContext<'tcx> {
         // we still want to screen for an "interesting" point to
         // highlight (e.g., a call site or something).
         let target_scc = self.constraint_sccs.scc(target_region);
-        let best_choice = (0..path.len()).rev().find(|&i| {
-            let constraint = path[i];
+        let mut range = 0..path.len();
+
+        let should_reverse = match from_region_origin {
+            NLLRegionVariableOrigin::FreeRegion
```

Comment:
I'm a *bit* surprised to see `FreeRegion` in this list, as these correspond to the lifetime parameters declared on the scope, and not higher-ranked subtyping per se -- which tests or scenarios led you to include it here?

Reply:
actually I was confused -- should-reverse being true is the *old* path, so this makes sense